src/libs/intents/printBase64Doc.ts

```diff
@@ -11,56 +11,66 @@ import { t } from '/locales/i18n'
 
 const log = MiniLog('intents:print')
 
-const safeDeleteFile = async (path: string): Promise<void> => {
-  try {
-    if (await RNFS.exists(path)) {
-      await RNFS.unlink(path)
-    }
-  } catch (err) {
-    log.error(
-      `Something went wront while deleting document to print: ${getErrorMessage(
-        err
-      )}`
-    )
-  }
-}
-
+// File Handling
 const getPrintFolderPath = (): string => {
   return `${RNFS.TemporaryDirectoryPath}/print/`
 }
 
-const getTemporaryPrintFilePath = (): string => {
+const getTemporaryPrintFilePath = (fileExtension: string): string => {
   const date = format(new Date(), 'yyyyMMddHHmmssSSS')
-  const tempFileName = `temp_print_${date}.printfile`
-
+  const tempFileName = `temp_print_${date}.${fileExtension}`
   const pathPrefix = Platform.OS === 'android' ? 'file://' : ''
-
   const path = `${pathPrefix}${getPrintFolderPath()}${tempFileName}`
-
   return path
 }
 
 const ensurePrintFolder = async (): Promise<void> => {
   await RNFS.mkdir(getPrintFolderPath())
 }
 
+const safeDeleteFile = async (path: string): Promise<void> => {
+  try {
+    if (await RNFS.exists(path)) {
+      await RNFS.unlink(path)
+    }
+  } catch (err) {
+    log.error(`Error deleting document: ${getErrorMessage(err)}`)
+  }
+}
+
+// Data URI Handling
+const getFileTypeFromDataURI = (dataURI: string): string => {
+  const mediaTypePattern = /^data:(.*?);base64,/
+  const match = dataURI.match(mediaTypePattern)
+  if (match?.[1]) {
+    const mediaType = match[1]
+    const parts = mediaType.split('/')
+    return parts[1]
+  }
+  throw new Error('Could not determine file type from data URI')
+}
+
+const getContentFromBase64 = (base64String: string): string => {
+  const dataUriPattern = /^data:([a-zA-Z]+\/[a-zA-Z]+);base64,/
+  return dataUriPattern.test(base64String)
+    ? base64String.split('base64,')[1]
+    : base64String
+}
+
+// Print Handling (main feature)
 export const printBase64Doc = async (base64?: string): Promise<void> => {
-  const path = getTemporaryPrintFilePath()
+  let path = ''
 
   try {
     if (!base64) throw new Error('No base64 provided')
+    path = getTemporaryPrintFilePath(getFileTypeFromDataURI(base64))
+    const fileContent = getContentFromBase64(base64)
 
     await ensurePrintFolder()
-    await RNFS.writeFile(path, base64, 'base64')
-
-    await RNPrint.print({ filePath: path })
+    await RNFS.writeFile(path, fileContent, 'base64')
+    await RNPrint.print({ filePath: path.replace('file://', '') })
```

Comment:
Why do you need to do a `path.replace` again ? It is already in `getTemporaryPrintFilePath` no ?

Reply:
Yes, need to remove it. With the prefix the print lib can't find the file. Maybe writeFile could not use it so it's clearer

Reply:
@zatteo to sum up: from what I understand I need the file prefix when writing the file, but when reading it with the print library, having that prefix will make the print software unable to read the file. I made it clearer by abstracting this file prefix in a "writeFile" local function

Reply:
For backup feature, it seems that I remove 'file://' both for iOS and Android.

I do not know what it is kept on o[cr feature / if it's needed here](https://github.com/cozy/cozy-flagship-app/blob/master/src/app/domain/ocr/services/ocr.ts#L61).